### PR TITLE
Merge cells in sheets #70

### PR DESCRIPTION
**Pull request by GregKapustin**
Merge cells per-sheet ; see issue #70

### CODE REVIEW

**Comment by GregKapustin** [comment]
Note: Codacy isn't happy because of the original lib/ files... let's ignore this.

**Comment by mdopark** [comment]
Looks great~!! But the example "Building a xlsx (spannig multiple rows  A1:A4 )" in Readme need to be changed also.

**Comment by yuflowdev** [comment]
Done. Plus if there's only one sheet and global option is set, I set it in sheet for retro-compatibility.

File: lib/index.js [hunk 1]
```diff
@@ -40,10 +39,11 @@ function build(worksheets) {
   var workBook = new _workbook2.default();
   worksheets.forEach(function (worksheet) {
     var name = worksheet.name || 'Sheet';
-    var data = (0, _helpers.buildSheetFromMatrix)(worksheet.data || [], worksheet.options || {});
+    var data = (0, _helpers.buildSheetFromMatrix)(worksheet.data || [], worksheet.options ? worksheet.options : worksheets.length === 1 ? options || {} : {}); // If worksheet has options, use this one ; if there's only one worksheet, use global options (for retro-compatibility)
```
Comment (cuspymd) [inline lib/index.js]: Looks good!! But do you think the worksheet.length comparison is necessary?
It would be enough that there is no worksheet.option.

Comment (yuflowdev) [reply lib/index.js]: The idea was "better no options than a merging occurring on all worksheets" (unless there is one worksheet).

As you wish guys :-)

Comment (GregKapustin) [reply lib/index.js]: The idea was "better no options than a merging occurring on all worksheets" (unless there is one worksheet).

As you wish guys :-)

**Comment by mgcrea** [comment]
You should never commit lib files, I build them before pushing a new release to npm. Please remove them from this PR. Thanks.

**Comment by GregKapustin** [comment]
No problem @mgcrea but how do I commit the changes in lib/index.js then ?

File: README.md [hunk 2]
```diff
@@ -38,7 +38,7 @@ const data = [[1, 2, 3], [true, false, null, 'sheetjs'], ['foo', 'bar', new Date
 var buffer = xlsx.build([{name: "mySheetName", data: data}]); // Returns a buffer
 ```
 
-  * Building a xlsx (spannig multiple rows `A1:A4`)
+  * Building a xlsx (spannig multiple rows `A1:A4` in all sheets)
```
Comment (cuspymd) [inline README.md]: "in all sheets" doesn't seem to be correct description.
As you changes, global merge option will be applied only given one sheet.

**Comment by cuspymd** [comment]
@GregKapustin It is enough to modify ```src/index.js```.

File: README.md (hunk 2 above)
Comment (GregKapustin) [reply README.md]: Done

**Comment by GregKapustin** [comment]
I need second argument to `buildSheetFromMatrix` to be the **worksheet's option** and **not the global options**.

How do we do it without modifying that line in lib/index.js ?
`const data = buildSheetFromMatrix(worksheet.data || [], options);`
to
`var data = (0, _helpers.buildSheetFromMatrix)(worksheet.data || [], worksheet.options`

**Comment by mdopark** [comment]
@GregKapustin ```lib/index.js``` is not a source code managed in Repository.
It is generated as a result of compiling ```src/index.js``` during the publish process.

**Pull request closed without merging**